### PR TITLE
Add Relationship Test.

### PR DESCRIPTION
**Pull request by KarinG**
Overview
----------------------------------------
Adding a First Test for Relationship and fixed a bug in the process.
This is also the First test that with multiple contacts. 

Before
----------------------------------------
No relationship test

After
----------------------------------------
Test! 🎉

### CODE REVIEW

File: src/Utils.php [hunk 1]
```diff
@@ -384,17 +384,17 @@ function wf_crm_get_contact_relationship_types($type_a, $type_b, $sub_type_a, $s
...
       // Reciprocal form - only show if different from above
       if ($reciprocal
         && ($t['type_a'] == $type_b || !$t['type_a'])
         && ($t['type_b'] == $type_a || !$t['type_b'])
-        && (in_array($t['sub_type_a'], $sub_type_b) || !$t['sub_type_a'])
-        && (in_array($t['sub_type_b'], $sub_type_a) || !$t['sub_type_b'])
+        && (in_array(strtolower($t['sub_type_a']), $sub_type_b) || !strtolower($t['sub_type_a']))
+        && (in_array(strtolower($t['sub_type_b']), $sub_type_a) || !strtolower($t['sub_type_b']))
       ) {
         $ret[$t['id'] . '_b'] = $t['label_b_a'];
```
Comment (KarinG) [inline src/Utils.php]: This is fixing a bug very similar to what we noticed in the `ContactDedupeTest` -> originally the Student - Organization relationship did not show up as an option b/c `Student` did not match `student`.

File: tests/src/FunctionalJavascript/ContactDedupeTest.php [hunk 2]
```diff
@@ -3,7 +3,6 @@
 namespace Drupal\Tests\webform_civicrm\FunctionalJavascript;
 
 use Drupal\Core\Url;
-use Drupal\FunctionalJavascriptTests\DrupalSelenium2Driver;
```
Comment (KarinG) [inline tests/src/FunctionalJavascript/ContactDedupeTest.php:6]: Tidy up.

**Comment by KarinG** [comment]
And ✅ 🎉

File: src/Utils.php (hunk 1 above)
Comment (jitendrapurohit) [reply src/Utils.php]: Do we always have `$sub_type_b` and `$sub_type_a` available in lower case? Given that in_array does a case sensitive comparison?

Comment (KarinG) [reply src/Utils.php]: A better solution would be to have an in_array non-case sensitive comparison of keys b/c we end up with `sub_type = Student` and `sub_type = student`, etc. We have seen that in a few places now. 

Implications are big: e.g. custom fields on Individual -> Student were not showing up (prior to using strtolower there); in this case you can not create a relationship between Individual -> Student and Organization b/c it does not think there are any relationships for Student (just for student).

I specifically created a sub_type -> Student (upper case) to test this. It's the most common scenario as well.

Comment (KarinG) [reply src/Utils.php]: Ok I've resolved this by ensuring we're always comparing contact sub_type in lower case 👆

Comment (KarinG) [reply src/Utils.php]: Will await test results before merging!

Comment (KarinG) [reply src/Utils.php]: Oh how did that fail? I need to take this back to my local.

Comment (KarinG) [reply src/Utils.php]: Ok I found a better place for the `strtolower()` 👍

File: src/Utils.php [hunk 3]
```diff
@@ -357,7 +357,13 @@ function wf_crm_get_relationship_types() {
...
         $r['type_b'] = strtolower(wf_crm_aval($r, 'contact_type_b'));
         $r['sub_type_a'] = wf_crm_aval($r, 'contact_sub_type_a');
+        if (!is_null($r['sub_type_a'])) {
+          $r['sub_type_a'] = strtolower($r['sub_type_a']);
+        }
         $r['sub_type_b'] = wf_crm_aval($r, 'contact_sub_type_b');
+        if (!is_null($r['sub_type_b'])) {
+          $r['sub_type_b'] = strtolower($r['sub_type_b']);
+        }
         $types[$r['id']] = $r;
```
Comment (KarinG) [inline src/Utils.php:367]: The `!is_null` is important here! If it is `null` it needs to remain `null`.

**Comment by KarinG** [comment]
These tests are passing on my local - fingers 🤞

**Comment by jitendrapurohit** [comment]
great thanks @KarinG 👍

**Pull request merged**